src/app/games/hive/hive.component.ts

```diff
@@ -0,0 +1,298 @@
+        this.inspectedStackTransform = `translate(${inspectedStackPosition.x} ${inspectedStackPosition.y})`;
+
+        const spaceForInspectedStack: number = this.SPACE_SIZE*5;
+        this.viewBox = boardAndRemainingViewBox
+            .expand(0, spaceForInspectedStack, 0, 0)
+            .toSVGString();
+    }
+
+    private getAllNeighbors(): Coord[] {
+        const neighbors: MGPSet<Coord> = new MGPSet();
+        for (const piece of this.pieces) {
+            neighbors.union(new MGPSet(this.getState().emptyNeighbors(piece.coord)));
+        }
+        if (neighbors.isEmpty()) {
+            // We need at least one clickable coord to be playable at first turn
+            neighbors.add(new Coord(0, 0));
+        }
+        return neighbors.toList();
+    }
+
+    public cancelMoveAttempt(): void {
+        this.selectedRemaining = MGPOptional.empty();
+        this.selectedStart = MGPOptional.empty();
+        this.selectedSpiderCoords = [];
+        this.selected = [];
+        this.indicators = [];
+        this.inspectedStack = MGPOptional.empty();
+    }
+
+    public getRemainingPieceTransformAsCoord(piece: HivePiece): Coord {
+        const shift: number = this.getRemainingPieceShift(piece);
+        const x: number = this.boardViewBox.center().x + shift * this.SPACE_SIZE * 4;
+        let y: number;
+        if (piece.owner === this.role) {
+            // Current player is below
+            y = this.boardViewBox.bottom() + (this.SPACE_SIZE * 3);
+        } else {
+            y = this.boardViewBox.up - (this.SPACE_SIZE * 2);
+        }
+        return new Coord(x, y);
+    }
+
+    public getRemainingPieceTransform(piece: HivePiece): string {
+        const transform: Coord = this.getRemainingPieceTransformAsCoord(piece);
+        return `translate(${transform.x} ${transform.y})`;
+    }
+
+    public getRemainingPieceHighlightTransform(piece: HivePiece): string {
+        const transform: Coord = this.getRemainingPieceTransformAsCoord(piece);
+        const size: number = this.getState().remainingPieces.getRemaining(piece);
+        return `translate(${transform.x} ${transform.y - (this.PIECE_HEIGHT * size)})`;
+    }
+
+    private getRemainingPieceShift(piece: HivePiece): number {
+        switch (piece.kind) {
+            case 'QueenBee': return -2.5;
+            case 'Beetle': return -1.5;
+            case 'Grasshopper': return -0.5;
+            case 'Spider': return 0.5;
+            default:
+                Utils.expectToBe(piece.kind, 'SoldierAnt', 'piece must be a soldier ant');
+                return 1.5;
+        }
+    }
+
+    public async selectRemaining(piece: HivePiece): Promise<MGPValidation> {
+        const clickValidity: MGPValidation = this.canUserPlay(`#remainingPiece_${piece.toString() }`);
+        if (clickValidity.isFailure()) {
+            return this.cancelMove(clickValidity.getReason());
+        }
+        if (piece.owner !== this.getCurrentPlayer()) {
```

Comment:
=== this.getCurrentOpponent() :p
wink wink nudge nudge

Reply:
commentaire doublon ?
Donc, on veut faire 
```
si (conditionInverseDeA)
  erreur(condition A est fausse)
```
plutôt que
```
si (conditionA === false) {
  erreur(condition A est fausse)
```
? C'pas contre-lecturable ?

Reply:
non juste 
```
if (conditionPourMeriterUnMessageDErreur) {
	erreur(laNatureDeLErreurExpliquée)
}
```
Et formuler le tout de façon la moins négativée

Reply:
fait, mais c'dangereux à généraliser, si on avait la possibilité d'avoir un troisième joueur (ou un observeur) ça casse !

Reply:
l'observeur serait pas un Player, et c'est pas trop prévu la dé-jscaipisation à priori :D !

Reply:
oui c'est vrai dans ce cas ! tpr donc ?